### PR TITLE
feat(cli): provide relevant info if snapshot operation is taking too long

### PR DESCRIPTION
**Pull request by y-lakhdar**
https://coveord.atlassian.net/browse/CDX-411

## Proposed changes
If the operation is taking too much time, the process will stop and display a message to help the user monitor his operation.
image

## Testing

- [ ] Unit Tests:
  
- [ ] Functionnal Tests:
  
- [x] Manual Tests:
By reducing the polling time

-----
CDX-411

### CODE REVIEW

File: packages/cli/src/lib/snapshot/snapshot.ts [hunk 1]
```diff
@@ -135,7 +136,7 @@ export class Snapshot {
         );
 
         if (isNotDone) {
-          throw new Error('Snapshot is still being processed');
+          throw new Error(this.operationGettingTooMuchTimeMessage());
```
Comment (olamothe) [inline packages/cli/src/lib/snapshot/snapshot.ts:139]: Is this an error though ? 🤔 

 My first reaction when I saw the message/screenshot, is that it's not bugged, it's "normal" that sometimes it takes a long time. 

So more like an "info" ?

**Review by olamothe — APPROVED** [review]

File: packages/cli/src/lib/snapshot/snapshot.ts (hunk 1 above)
Comment (olamothe) [reply packages/cli/src/lib/snapshot/snapshot.ts:139]: Related to that, what would be missing is something that explains: 

"After you run monitor command and it finally finishes, the next step would be to run XYZ".

File: packages/cli/src/lib/snapshot/snapshot.ts [hunk 2]
```diff
@@ -135,7 +137,7 @@ export class Snapshot {
         );
 
         if (isNotDone) {
-          throw new Error('Snapshot is still being processed');
+          throw new Error(SnapshotErrors.operationTimedOut);
```
Comment (louis-bompart) [inline packages/cli/src/lib/snapshot/snapshot.ts:140]: I think exponentialBackoff here does not fit properly our needs compared to async-retry, I'll get you a suggestion built up 🔜:tm:

File: packages/cli/src/lib/snapshot/snapshot.ts (hunk 1 above)
Comment (y-lakhdar) [reply packages/cli/src/lib/snapshot/snapshot.ts]: I guess this help message would only be shown within the context of a preview command. In this case, the next step would be to run `org:config:push`.

However, what would be the next step of a `org:config:push`?

**Review by louis-bompart — APPROVED** [review]

**Pull request merged**